tests/main/refresh-classic/task.yaml

```diff
@@ -0,0 +1,38 @@
+summary: Check that refresh from classic->strict works
+
+# ubuntu-core does not support classic confinement
+systems: [-ubuntu-core-*]
+
+prepare: |
+    #shellcheck source=tests/lib/dirs.sh
+    . "$TESTSLIB"/dirs.sh
+
+    case "$SPREAD_SYSTEM" in
+        fedora-*|arch-*|centos-*)
+            # although classic snaps do not work out of the box on fedora,
+            # we still want to verify if the basics do work if the user
+            # symlinks /snap to $SNAP_MOUNT_DIR themselves
+            ln -sf $SNAP_MOUNT_DIR /snap
```

Comment:
I didn't check this deeply but IIRC we provide this symlink in global prepare. This code doesn't check for absence of the symlink. The restore section should be conditional on the initial absence of the symlink

Reply:
I just double checked in spread:
```
google:fedora-31-64 .../tests/main/refresh-classic# ls -l /snap
ls: cannot access '/snap': No such file or directory
```

Reply:
Thanks for checking. I think I would still prefer to be more careful here. If you don't mind I can either change the code in this PR or we can merge it and I'll send a follow-up.

Reply:
Thanks, this pattern is used in a bunch of other tests too, I cargo culted it :( Let's do a followup that move this into some sort of helper where we can then DTRT in a single place.